static/js/homepage/carousel.tsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component that shows items in a carousel.
+ */
+
+import React, { useState } from "react";
+
+interface CarouselItem {
+  id: string;
+  url: string;
+  title: string;
+  imgUrl: string;
+}
+
+interface CarouselPropType {
+  items: CarouselItem[];
+}
+
+export function Carousel(props: CarouselPropType): JSX.Element {
```

Comment:
I didn't think to look into libraries & it was simple enough to implement on our own & gives more control for customization

Reply:
carousels usually have the swiping animations and left right arrow show hide logic there. wonder do you need the left arrow after swiping to the end?

Reply:
I don't think a left arrow will be needed because this will just continue again to the start .. can also switch to a library in the future if requirements get more complicated for this component

![screen-recording-_97_](https://github.com/datacommonsorg/website/assets/69875368/bad913d2-ad9d-462f-81f9-963dec578059)